llvm/lib/Target/AIE/AIE2AddrSpace.h

```diff
@@ -40,6 +42,71 @@ enum class AddressSpaces {
 
 enum class AIEBanks { A, B, C, D, TileMemory };
 
+static inline uint64_t getDefaultMemoryBank() {
+  std::bitset<32> MemoryBanks;
+  MemoryBanks.set(static_cast<unsigned>(AIEBanks::A))
+      .set(static_cast<unsigned>(AIEBanks::B))
+      .set(static_cast<unsigned>(AIEBanks::C))
+      .set(static_cast<unsigned>(AIEBanks::D));
+  return MemoryBanks.to_ulong();
+}
+
+static inline uint64_t getMemoryBanksFromAddressSpace(unsigned AddrSpace) {
```

Comment:
I got your design situation. The pass you are creating is a fully IR pass, so you cannot have access to some backend parts, like subtarget, instruction info, for example. In this way you can have the missing information from this file.